### PR TITLE
Fixes #2664 ItemAttribute types in constants

### LINKED ISSUE #2664: Item Attributes type definition is unreadable
This code ( https://github.com/otland/forgottenserver/blob/master/src/item.h#L499-L504 ):
```
		static bool isIntAttrType(itemAttrTypes type) {
			return (type & 0x7FFE13) != 0;
		}
		static bool isStrAttrType(itemAttrTypes type) {
			return (type & 0x1EC) != 0;
		}
```
Must be changed to some readable format. I wasted 30 minutes on debuging my code.

Just because I forgot there is this s***y number and my new attribute must be added there (with calculator). Otherwise it won't work.

It should be definied as some variable:
```
uint32_t INT_ATTRIBUTES = ITEM_ATTRIBUTE_ACTIONID | ITEM_ATTRIBUTE_UNIQUEID | (...);
uint32_t STRING_ATTRIBUTES = ITEM_ATTRIBUTE_NAME | ITEM_ATTRIBUTE_TEXT | (...);
```` 
I will create PR when I have some free time.

// gesior at work

### PR DESCRIPTION
**Pull request by gesior**
Moved `itemAttrTypes` to constants to make it more readable.

Also fixed comparing attribute types in functions `isIntAttrType`, `isStrAttrType` and `isCustomAttrType` as it was possible to pass not existing type to it (from Lua).

Fixes #2664

### CODE REVIEW

File: src/item.h [hunk 1]
```diff
@@ -495,15 +495,24 @@ class ItemAttributes
...
 		}
 
+		const static uint32_t intAttrType = ITEM_ATTRIBUTE_ACTIONID | ITEM_ATTRIBUTE_UNIQUEID | ITEM_ATTRIBUTE_DATE
+			| ITEM_ATTRIBUTE_WEIGHT | ITEM_ATTRIBUTE_ATTACK | ITEM_ATTRIBUTE_DEFENSE | ITEM_ATTRIBUTE_EXTRADEFENSE
+			| ITEM_ATTRIBUTE_ARMOR | ITEM_ATTRIBUTE_HITCHANCE | ITEM_ATTRIBUTE_SHOOTRANGE | ITEM_ATTRIBUTE_OWNER
+			| ITEM_ATTRIBUTE_DURATION | ITEM_ATTRIBUTE_DECAYSTATE | ITEM_ATTRIBUTE_CORPSEOWNER | ITEM_ATTRIBUTE_CHARGES
+			| ITEM_ATTRIBUTE_FLUIDTYPE | ITEM_ATTRIBUTE_DOORID;
+		const static uint32_t stringAttrType = ITEM_ATTRIBUTE_DESCRIPTION | ITEM_ATTRIBUTE_TEXT | ITEM_ATTRIBUTE_WRITER
+			| ITEM_ATTRIBUTE_NAME | ITEM_ATTRIBUTE_ARTICLE | ITEM_ATTRIBUTE_PLURALNAME;
+		const static uint32_t customAttrType = ITEM_ATTRIBUTE_CUSTOM;
```
Comment (DSpeichert) [inline src/item.h]: This is not necessary, I doubt there will be more custom attribute types.

**Comment by DSpeichert** [comment]
This now needs to be rebased after #2667 please

File: src/item.h (hunk 1 above)
Comment (DSpeichert) [reply src/item.h]: Actually, stringAttrType and intAttrType should be plural: `stringAttributeTypes` and `intAttributeTypes` (yes, no need to abbreviate attr either).

Comment (gesior) [reply src/item.h]: Renamed to plural and full 'attribute' name. Removed 'custom' constant.

**Comment by gesior** [comment]
@DSpeichert 
Renamed constants, removed 'custom' constant, rebased to master and added new attribute that was added in master changes.

**Pull request merged**